src/Server/Handler/Request/CallToolHandler.php

```diff
@@ -58,7 +59,9 @@ public function handle(Request $request, SessionInterface $session): Response|Er
         $this->logger->debug('Executing tool', ['name' => $toolName, 'arguments' => $arguments]);
 
         try {
-            $reference = $this->registry->getTool($toolName);
+            // Try dynamic tool first, fall back to static
+            $reference = $this->registry->getDynamicTool($toolName)
+                ?? $this->registry->getTool($toolName);
```

Comment:
This means we only get a Tool if there are no dynamic Tools. Wouldn't that be cause for a lot of confusion?